### PR TITLE
Update RTL plugin callback description

### PR DESCRIPTION
**Pull request by HeyStenson**
fixes https://github.com/mapbox/mapbox-gl-js/issues/9689

@ryanhamley I think I'm butchering [the point you made about how this callback actually works](https://github.com/mapbox/mapbox-gl-js/issues/9689#issuecomment-628950203) - could you help me update the wording for the `callback` param for [`setRTLTextPlugin`](https://docs.mapbox.com/mapbox-gl-js/api/properties/#setrtltextplugin)?

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -257,7 +257,7 @@ Debug.extend(exported, {isSafari, getPerformanceMetrics: PerformanceUtils.getPer
  *
  * @function setRTLTextPlugin
  * @param {string} pluginURL URL pointing to the Mapbox RTL text plugin source.
- * @param {Function} callback Called with an error argument if there is an error.
+ * @param {Function} callback Called with an error argument if there is an error, or a null error argument on success.
```
Comment (ryanhamley) [inline src/index.js:260]: I think it would be better to say `, or no arguments if the plugin loads successfully`

**Review by ryanhamley — CHANGES_REQUESTED** [review]
@HeyStenson I clarified it a touch but you have the gist correct. The callback will always be called either with no arguments if the plugin loads successfully or with an error argument if an error occurs.

As noted in https://github.com/mapbox/mapbox-gl-js/issues/9689#issuecomment-628958099, the callback is typed as an `ErrorCallback` which isn't true. I'm not sure if that used to be the case, but it's clearly not anymore. It's pretty easy to fix this though since we already have a generic `Callback` type. 

In `rtl_text_plugin.js`, we'd need to update the following:

1. `import type {Callback} from '../types/callback.js';`
2. Delete this line https://github.com/mapbox/mapbox-gl-js/blob/4a92eba39cae7c7fc8e2db13d5d6e35ae96c0ee9/src/source/rtl_text_plugin.js#L22
3. Update the type of `callback` in this line to be `?Callback<{err: ?Error}>` https://github.com/mapbox/mapbox-gl-js/blob/4a92eba39cae7c7fc8e2db13d5d6e35ae96c0ee9/src/source/rtl_text_plugin.js#L64

**Comment by HeyStenson** [comment]
Thanks @ryanhamley! I made these changes - ready for you to take another look.

**Review by ryanhamley — APPROVED** [review]
Looks good! Thanks for making those changes @HeyStenson

**Pull request merged**